### PR TITLE
add get by username and get by wallet addr

### PR DESCRIPTION
**Pull request by bennycio**
Changes:

- **Added** parts of get func to handle getting by address or username
- **Changed** get user input to reflect possible query values
- **Added** db func for getting by username

Considerations:

- Currently there is no validation to ensure that the query only contained one of: username, wallet, user_id -- and instead will only handle once based on their priority. This priority is user_id, username, address (in order of least to most complex/time consuming query). That means the query could be something like /glry/v1/users/get?username=:username&addr=:walletAddress&user_id=:userId and it would still work. Do we want it to function this way, or do we want to ensure that only one query param is given?

### CODE REVIEW

**Comment by bennycio** [comment]
Somehow I didn't even mention that this is for getting **users** if that was not clear.

File: persist/user.go [hunk 1]
```diff
@@ -109,10 +117,44 @@ func UserGetByAddress(pAddress string,
...
+	mp := NewMongoStorage(0, usersCollName, pRuntime)
+
+	result := []*User{}
+	err := mp.Find(pCtx, bson.M{"addresses": bson.M{"$in": []string{pAddress}}}, result, opts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result) == 0 || len(result) > 1 {
```
Comment (fbrcsftnr) [inline persist/user.go:135]: Can we break this into two separate errors? i.e. "no users found with wallet address" and "greater than 1 user found with wallet address"

File: server/user.go [hunk 2]
```diff
@@ -245,16 +225,38 @@ func userGetDb(pInput *userGetInput,
...
+			return nil, err
+		}
+		break
+	case pInput.Address != "":
+		user, err = persist.UserGetByAddress(pInput.Address, pCtx, pRuntime)
+		if err != nil {
+			return nil, err
+		}
+		break
+	}
```
Comment (fbrcsftnr) [inline server/user.go:248]: Beautiful

**Comment by fbrcsftnr** [comment]
Awesome cleanup!

Re: priority/validation of query params, it's fine if the client sends multiple in my opinion, since it's sort of on them if they decide to include a user ID + wallet pair that's invalid

**Review by fbrcsftnr — APPROVED** [review]

File: persist/user.go (hunk 1 above)
Comment (bennycio) [reply persist/user.go]: sounds good, reflected on most recent commit!

**Pull request merged**